src/tests/sqlsmith/src/sqlreduce/passes/pullup.rs

```diff
@@ -54,8 +54,8 @@ impl Transform for BinaryOperatorPullup {
         reduction_points
     }
 
-    fn apply_on(&self, ast: &mut Ast, reduction_points: Vec<usize>) -> Ast {
-        if let Some(query) = extract_query_mut(ast)
+    fn apply_on(&self, ast: Ast, reduction_points: Vec<usize>) -> Ast {
+        if let Some(mut query) = extract_query_mut(ast.clone())
```

Comment:
The change in `extract_query_mut` is incorrect. We want to modify the query of the original ast, however, here the original ast is not changed. Only the intermediate `query` is modified, and then dropped.

The correct should be like
```
fn apply_on(&self, mut ast: Ast, reduction_points: Vec<usize>) -> Ast {
        if let Some(mut query) = extract_query_mut(&mut ast) {
            ...
        }
        ast
```

with `extract_query_mut` unchanged.